packages/kuma-gui/src/app/vue/index.ts

```diff
@@ -45,19 +48,45 @@ export const services = (app: Record<string, Token>): ServiceDefinition[] => {
   return [
     [$.app, {
       service: (
+        plugins: PluginDefinition[],
         components: ComponentDefinition[],
         directives: DirectiveDefinition[],
-        plugins: PluginDefinition[],
+        componentWalkers: ComponentWalkerDefinition[],
       ) => {
         return async (App: Component) => {
           const app = createApp(App)
+          const app_component = app.component.bind(app)
+
+          const proxyApp = new Proxy(app, {
+            get(target, prop: keyof VueApp) {
+              switch(prop) {
+                case 'use':
+                  return (...[plugin, options]: PluginDefinition) => {
+                    if(typeof plugin.install === 'function') {
+                      plugin.install(proxyApp, options)
+                      return proxyApp
+                    }
+                  }
+                case 'component':
+                  return (...args: ComponentDefinition | [string]) => {
+                    if(args.length === 1) {
+                      return app_component(...args)
+                    }
+                    app_component(...componentWalkers.reduce((prev, item) => item(...prev), args))
+                    return proxyApp
+                  }
+                default:
+                  return target[prop]
+              }
+            },
+          })
```

Comment:
This is the "meat" of the PR.

Instead of installing components when a plugin is installed, call a function on the component definition passed in _before_ actually installing it with Vue.

This gives us the opportunity to amend it if we need to.

This is very very similar to how our containers decorators work, but instead of doing this at "container build time" we do it at "runtime/component installation time"